lib/scholar/neighbors/kd_tree.ex

```diff
@@ -311,8 +333,8 @@ defmodule Scholar.Neighbors.KDTree do
     end
   end
 
-  defnp predict_n(tree, point, opts) do
-    k = opts[:k]
+  defnp predict_n(tree, point) do
+    k = tree.num_neighbors
```

Comment:
I we now pass `num_neighbors` in fit, I think we may add a note that there is no need to compute all KDTree from scratch for a different number of nearest neighbors.

Reply:
Perhaps, but then we do the same in `BruteKNN` and `RandomProjectionForest`. They all now take `num_neighbors` as an option to `fit`.